Jasmine/Common/Models/GameDataFactory.swift

```diff
@@ -23,10 +23,15 @@ class GameDataFactory {
     ///
     /// - Parameter difficulty: difficulty of the game
     /// - Returns: game data containing relevant phrases
-    func createGame(difficulty: Int) -> GameData {
+    func createGame(difficulty: Int, type: GameType) -> GameData {
         let gameData = GameData()
         gameData.difficulty = difficulty
-        gameData.phrases = realm.objects(Phrase.self)
+        switch type {
+        case .chengYu:
+            gameData.phrases = realm.objects(Phrase.self).filter("chinese LIKE '????'")
```

Comment:
I'm not sure really, why not maybe just create 2 separate tables for phrases and chengyus? In the name of extensibility? haha